ServiceTier/Services/ITaskService.cs

```diff
@@ -0,0 +1,26 @@
+﻿using Common.ModelsDTO;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace ServiceTier.Services
+{
+    public interface ITaskService : IBaseService
+    {
+        IQueryable<TaskDTO> GetAllTasks();
+        IQueryable<TaskDTO> GetAllTasksByUserId(int userId);
+        IQueryable<TaskDTO> GetAllActiveTasks();
+        IQueryable<TaskDTO> GetAllActiveTasksByUserId(int userId);
+        IQueryable<TaskDTO> GetAllArchivedTasks();
+        IQueryable<TaskDTO> GetAllArchivedTasksByUserId(int userId);
+        TaskDTO GetTaskById(int taskId);
+        void CreateTask(TaskDTO taskDTO);
+        void UpdateTaskById(TaskDTO taskDTO);
+        void DeleteTaskById(int taskId);
+        void UpdateTaskStatus(int taskId, bool newStatus);
+        void ResetTaskById(int taskId);
+        void StartTimer(int taskId);
+        void PauseTimer(int taskId);
+        //TODO: saving to .xml and/or .csv file
```

Comment:
I don't like comments like this. Code is not a proper place for "todos" - ticket in bug tracing system is.